.github/workflows/build.yml

```diff
@@ -0,0 +1,259 @@
+          - {go: "${{ needs.get-go-version.outputs.go-version }}", goos: "windows", goarch: "amd64", component: "control-plane", pkg_name: "consul-k8s-control-plane" }
+      fail-fast: true
+
+    name: Go ${{ matrix.go }} ${{ matrix.goos }} ${{ matrix.goarch }} ${{ matrix.component }} build    
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go }}
+
+      - name: Build
+        env:
+          GOOS: ${{ matrix.goos }}
+          GOARCH: ${{ matrix.goarch }}
+          CGO_ENABLED: 0
+        working-directory: ${{ matrix.component }}
+        run: |
+          mkdir -p dist out
+          GCO_ENABLED=0 go build -o dist/${{ matrix.pkg_name }} .
+          zip -r -j out/${{ matrix.pkg_name }}_${{ needs.get-product-version.outputs.product-version }}_${{ matrix.goos }}_${{ matrix.goarch }}.zip dist/
+
+      - name: Upload built binaries 
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.pkg_name }}_${{ needs.get-product-version.outputs.product-version }}_${{ matrix.goos }}_${{ matrix.goarch }}.zip
+          path: ${{ matrix.component}}/out/${{ matrix.pkg_name }}_${{ needs.get-product-version.outputs.product-version }}_${{ matrix.goos }}_${{ matrix.goarch }}.zip
+            
+      - name: Package rpm and deb files 
+        if: ${{ matrix.goos == 'linux' && matrix.component == 'cli' }}
+        uses: hashicorp/actions-packaging-linux@v1
+        with:
+          name: consul-k8s 
+          description: "consul-k8s provides a cli interface to first-class integrations between Consul and Kubernetes."
+          arch: ${{ matrix.goarch }}
+          version: ${{ needs.get-product-version.outputs.product-version }}
+          maintainer: "HashiCorp"
+          homepage: "https://github.com/hashicorp/consul-k8s"
+          license: "MPL-2.0"
+          binary: "${{ matrix.component }}/dist/${{ matrix.pkg_name }}"
+          deb_depends: "openssl"
+          rpm_depends: "openssl"
+
+      - name: Set package names
+        if: ${{ matrix.goos == 'linux' && matrix.component == 'cli' }}
+        run: |
+          echo "RPM_PACKAGE=$(basename out/*.rpm)" >> $GITHUB_ENV
+          echo "DEB_PACKAGE=$(basename out/*.deb)" >> $GITHUB_ENV
+
+      - name: Upload rpm packages 
+        uses: actions/upload-artifact@v2
+        if: ${{ matrix.goos == 'linux' && matrix.component == 'cli' }}
+        with:
+          name: ${{ env.RPM_PACKAGE }}
+          path: out/${{ env.RPM_PACKAGE }}
+
+      - name: Upload debian packages 
+        uses: actions/upload-artifact@v2
+        if: ${{ matrix.goos == 'linux' && matrix.component == 'cli' }}
+        with:
+          name: ${{ env.DEB_PACKAGE }}
+          path: out/${{ env.DEB_PACKAGE }}
+
+  build-darwin:
```

Comment:
could it be part of the `build` job?

Reply:
Oh yes, I think it could. I was following the pattern that I had seen other teams use but there really isn't any reason for it to be separate.

I will move it over.